src/llmcompressor/transformers/finetune/data/base.py

```diff
@@ -106,142 +169,144 @@ def get_raw_dataset(self, cache_dir: Optional[str] = None) -> Dataset:
+
+        return self.dataset_template
 
-            # store unpadded prompt so we can mask out correct number of elements
-            # in the labels
-            if self.PROMPT_KEY in data:
-                result[self.PROMPT_KEY] = self.tokenizer(
-                    data[self.PROMPT_KEY],
-                    max_length=self.max_seq_length,
-                    truncation=True,
-                )["input_ids"]
-
-            return result
-
-        # helper fn for filling to max_sequence_length by concatenating entries
-        def group_text_fn(data):
-            concatenated_data = {k: sum(data[k], []) for k in data.keys()}
-            total_length = len(concatenated_data[list(data.keys())[0]])
-            total_length = (total_length // self.max_seq_length) * self.max_seq_length
-            result = {
-                k: [
-                    t[i : i + self.max_seq_length]
-                    for i in range(0, total_length, self.max_seq_length)
-                ]
-                for k, t in concatenated_data.items()
-            }
-            return result
-
-        # helper fn for adding labels, needed for loss calculation
-        def label_fn(data):
-            # if the dataset uses prompts, mask them out so they don't contribute
-            # to the loss calculation
-            prompt_len = 0
-            if self.PROMPT_KEY in data:
-                prompt_len = len(data[self.PROMPT_KEY])
-            data["labels"] = data["input_ids"].copy()
-            data["labels"][:prompt_len] = [LABELS_MASK_VALUE] * prompt_len
-
-            # mask out padding in the labels as well
-            padding = len(data["attention_mask"]) - sum(data["attention_mask"])
-            if padding > 0:
-                data["labels"][-padding:] = [LABELS_MASK_VALUE] * padding
-            return data
-
-        if raw_dataset is None:
-            raw_dataset = self.get_raw_dataset()
-
-        dataset = self.map(
-            raw_dataset,
-            function=tokenize_fn,
-            batched=True,
-            remove_columns=[self.text_column],
-            num_proc=self.data_args.preprocessing_num_workers,
-            load_from_cache_file=not self.data_args.overwrite_cache,
-            desc="Running tokenizer on dataset",
+    @property
+    def dataset_template(self) -> Union[Callable[[Any], Any], None]:
+        return None
+
+    def rename_columns(self, dataset: DatasetType) -> DatasetType:
+        # rename columns to match processor/tokenizer kwargs
+        column_names = get_column_names(dataset)
+        if self.data_args.text_column in column_names and "text" not in column_names:
+            dataset = dataset.rename_column(self.data_args.text_column, "text")
+
+        return dataset
+
+    def filter_tokenizer_args(self, dataset: DatasetType) -> DatasetType:
+        # assumes that inputs are not passed via self.processor.__call__ args and kwargs
+        signature = inspect.signature(self.processor.__call__)
```

Comment:
As we talked offline, it would be nice to have some debug logs (not visible by default) so show what keys existed before, what we expect, and what we end up with during all of this mapping and filtering

Reply:
Added, see example in PR description. Thanks for the suggestion